modal/functions.py

```diff
@@ -338,6 +339,40 @@ def _bind_method(
     ):
         """mdmd:hidden
 
+        Creates a _Function that is bound to a specific class method name. This _Function is not uniquely tied
+        to any backend function -- its object_id is the function ID of the class service function.
```

Comment:
I don't think the new "methods" should even have an object id or be fully fledged `_Function`s (apart from their functionality).

The outline of what I'd like to do is roughly:

* Pull out all the user-callable `_Function` functionality (.remote, .map etc.) into its own base class `_AbstractFunctionActions` with an abstract `_get_invocation_metadata() -> (use_object_id, use_method_name)` method that's used by all places that make function calls etc.
* Make both `_Function` and `_MethodHandle` inherit from `_AbstractFunctionActions` and override `_get_invocation_metadata` in their own way
* Class/instance service functions would be `_Function`s (with object ids and load-hydration etc.) but class methods would be "dumb" `MethodHandle` objects that "hydrate" using data from their service functions.

I think a structure like that would let us separate more of the class-specific from the function logic to make the latter less complicated and fragile.